### ISSUE TITLE
wrong identity_conversion

### ISSUE THREAD

**Issue by Geobert** [rust-lang/rust-clippy#4207]
clippy 0.0.212 (265318db 2019-05-17)

I might be wrong, but it seems there's an issue here:

```rust
fn walk_dir(path: &Path) -> std::io::Result<()> {
    for entry in path.read_dir()? {
        if let Ok(entry) = entry {
            if entry.path().is_file() {
                do_stuff_on_file(&entry.path())?;
            } else if entry.path().is_dir() {
                walk_dir(&entry.path())?;
            }
        }
    }
    Ok(())
}
```

```
warning: identical conversion
  --> src\main.rs:41:18
   |
41 |     for entry in path.read_dir()? {
   |                  ^^^^^^^^^^^^^^^^ help: consider removing `path.read_dir()?()`: `path.read_dir()?`
   |
   = note: #[warn(clippy::identity_conversion)] on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_conversion
```

**Comment by Geobert** [comment]
my bad
duplicate #4133
